### PR TITLE
Flutter for Modroid should use vsync

### LINKED ISSUE #589: Block's height=double.INFINITY requirement can be confusing
Using a Block to layout a column of widgets fails if the Block's height is constrained.

See the comment in the example below.

```
import 'package:sky/rendering/box.dart';
import 'package:sky/theme/colors.dart';
import 'package:sky/widgets/basic.dart';
import 'package:sky/widgets/icon.dart';
import 'package:sky/widgets/scaffold.dart';
import 'package:sky/widgets/theme.dart';
import 'package:sky/widgets/tool_bar.dart';
import 'package:sky/widgets/framework.dart';
import 'package:sky/widgets/task_description.dart';

class TestApp extends App {
  static const green = const Color(0x8000FF00);

  void handleTap(Widget body, Widget problemChild) {
    print("TAP body: body: ${body.root.parentData.position & body.root.size}");
    print("TAP problemChild: ${problemChild.root.parentData.position & problemChild.root.size}");
  }

  Widget build() {
    // Expected the Block to be <parent-width>x<parent-height> at 0,0
    // Expected problemChild to be <parent-width>x100 at 0,0
    // This causes an assert because the Block's height isn't infinity:
    // E/chromium(27380): 'http://localhost:9888/packages/sky/rendering/block.dart': Failed assertion: line 151 pos 12: '_isVertical ? constraints.maxHeight >= double.INFINITY : constraints.maxWidth >= double.INFINITY' is not true.
    Widget problemChild = new Block([
      new Container(
        decoration: new BoxDecoration(backgroundColor: green),
        height: 100.0
      )
    ]);

    // everything that follows is (IMHO) app boilerplate

    Widget body;
    body = new Listener(
      onGestureTap: (e) { handleTap(body, problemChild); },
      child: new Container(
        decoration: new BoxDecoration(backgroundColor: Theme.of(this).cardColor),
        child: problemChild
      )
    );

    return new IconTheme(
      data: const IconThemeData(color: IconThemeColor.white),
      child: new Theme(
        data: new ThemeData(
          brightness: ThemeBrightness.light,
          primarySwatch: Blue,
          accentColor: RedAccent[200]
        ),
        child: new TaskDescription(
          label: 'Test',
          child: new Scaffold(
            toolbar: new ToolBar(center: new Text('Progress Indicators')),
            body: body
          )
        )
      )
    );
  }
}

void main() {
  runApp(new TestApp());
}
```

### PR DESCRIPTION
**Pull request by abarth**
This patch teaches Flutter how to connect to the vsync service on Modroid.
Also, remove our now-redunant copy of vsync.mojom.

Fixes #589

### CODE REVIEW

**Comment by abarth** [comment]
@krisgiesing @Hixie

File: sky/shell/ui/engine.cc [hunk 1]
```diff
@@ -120,12 +120,13 @@ void Engine::SetServices(ServicesDataPtr services) {
   services_ = services.Pass();
 
 #if defined(OS_ANDROID) || defined(OS_IOS)
-  if (services_->services_provided_by_embedder) {
-    // TODO(abarth): Implement VSyncProvider on other platforms.
-    vsync::VSyncProviderPtr vsync_provider;
+  vsync::VSyncProviderPtr vsync_provider;
+  if (services_->shell) {
```
Comment (krisgiesing) [inline sky/shell/ui/engine.cc]: I'm a little unclear on the significance of this if check. When is the shell present vs. not present?

**Comment by krisgiesing** [comment]
LGTM, my question is just me wanting to understand the system better.

File: sky/shell/ui/engine.cc (hunk 1 above)
Comment (abarth) [inline sky/shell/ui/engine.cc]: The shell is present when run in MojoShell and absent when run in SkyShell.

**Pull request merged**